inputs/range.vue

```diff
@@ -0,0 +1,411 @@
+      };
+    },
+    computed: {
+      label() {
+        return label(this.name, this.schema);
+      },
+      start() {
+        return this.args.start || 0;
+      },
+      isDualPoint() {
+        return _.isArray(this.start);
+      },
+      step() {
+        return this.args.step || 1;
+      },
+      min() {
+        return this.args.min || 0;
+      },
+      max() {
+        return this.args.max || 10;
+      },
+      minLabel() {
+        return this.args.minLabel || this.min.toString();
+      },
+      maxLabel() {
+        return this.args.maxLabel || this.max.toString();
+      },
+      tooltips() {
+        return !!this.args.tooltips;
+      },
+      errorMessage() {
+        return getValidationError(this.data, this.args.validate, this.$store, this.name);
+      },
+      showError() {
+        return !!this.errorMessage;
+      },
+      hasFeedback() {
+        return this.args.help || this.showError;
+      },
+      rangeClasses() {
+        return {
+          'has-tooltips': this.args.tooltips
+        };
+      }
+    },
+    methods: {
+      update(values) {
+        // if we're dealing with a single value, grab it directly
+        // otherwise grab the whole array
+        const val = this.isDualPoint ? values : _.head(values);
+
+        this.$store.commit(UPDATE_FORMDATA, { path: this.name, data: val });
+      }
+    },
+    mounted() {
+      const el = find(this.$el, '.editor-range-input'),
+        step = this.step,
+        min = this.min,
+        max = this.max,
+        minLabel = this.minLabel,
+        maxLabel = this.maxLabel;
+
+      slider.create(el, {
+        start: this.start,
+        step: step,
+        range: { min, max },
+        format: {
+          to(val) {
+            return step >= 1 ? parseInt(val) : val;
+          },
+          from(val) {
+            return step >= 1 ? parseInt(val) : val;
+          }
+        },
+        connect: this.isDualPoint ? [false, true, false] : [true, false],
+        tooltips: this.tooltips,
+        pips: {
+          mode: 'steps',
+          filter(val, type) {
+            if (!type || val % step !== 0) {
+              return -1; // only show pips for each step
+            }
+
+            // make sure we always show the min and max values
+            if (val === min) {
+              return 2;
+            }
+
+            if (val === max) {
+              return 2;
+            }
+
+            // use small pips for each step between the min and max values
+            if (val % step === 0) {
+              return 0;
+            }
+          },
+          format: {
+            to(val) {
+              if (val === min) {
+                return minLabel;
+              }
+
+              if (val === max) {
+                return maxLabel;
+              }
+            }
+          }
+        }
+      });
+
+      // add keyboard support
+      _.each(findAll(el, '.noUi-handle'), (handle, index) => {
```

Comment:
consider native JS here

Reply:
NodeLists aren't arrays, so IE in particular doesn't have a native `forEach` method on their prototype 😄

Reply:
that's true for IE but you can call the prototype in the contexts of a NodeList, if we're using the Babel polyfill it adds this to NodeList and we don't support IE anyway.

Reply:
I've found that `Array.from(NodeList).forEach` works natively